### PR TITLE
Delete single reaction

### PR DESCRIPTION
**Pull request by Skillz4Killz**
This PR adds the functionality to delete all reactions for a single emoji.

### CODE REVIEW

**Comment by Khaaz** [comment]
ref: discordapp/discord-api-docs#1309

File: lib/gateway/Shard.js [hunk 1]
```diff
@@ -594,6 +594,23 @@ class Shard extends EventEmitter {
...
+                if(!channel) {
+                    break;
+                }
+                /**
+                * Fired when someone removes all reactions from a message for a single emoji
+                * @event Client#messageReactionRemoveAllForEmoji
+                * @prop {Message | Object} message The message object. If the message is not cached, this will be an object with `id` and `channel` keys. No other property is guaranteed
+                * @prop {Object} emoji The emoji object with a `name` prop. If the emoji is a custom emoji it will also have an `id` prop.
+                */
+                this.emit("messageReactionRemoveAllForEmoji", channel.messages.get(packet.d.message_id) || {
```
Comment (Brayzure) [inline lib/gateway/Shard.js]: Event name should probably be `messageReactionRemoveEmoji`. Not as wordy, and it matches the payload event name

File: lib/structures/Message.js [hunk 2]
```diff
@@ -333,6 +333,15 @@ class Message extends Base {
         return this._client.removeMessageReactions.call(this._client, this.channel.id, this.id);
     }
 
+    /**
+    * Remove all reactions from a message for a single emoji
+    * @arg {String} reaction The reaction (Unicode string if Unicode emoji, `emojiName:emojiID` if custom emoji)
+    * @returns {Promise}
+    */
+    removeReactionsForEmoji(reaction) {
```
Comment (Brayzure) [inline lib/structures/Message.js]: This method name should match what's present in the client.

**Review by Brayzure — CHANGES_REQUESTED** [review]

File: lib/structures/Message.js (hunk 2 above)
Comment (Skillz4Killz) [reply lib/structures/Message.js]: @Brayzure This is the same name following the consistency in the other methods. The other methods also remove `Message` from their names so i had kept it that way as well.

Comment (Brayzure) [reply lib/structures/Message.js]: Oh true, I do see that.

**Review by Brayzure — DISMISSED** [review]

File: lib/structures/Message.js (hunk 2 above)
Comment (abalabahaha) [inline lib/structures/Message.js]: Seems to be missing typings

File: lib/rest/Endpoints.js [hunk 3]
```diff
@@ -12,6 +12,7 @@ module.exports.CHANNEL_INVITES =                                        (chanID)
 module.exports.CHANNEL_MESSAGE_REACTION =              (chanID, msgID, reaction) => `/channels/${chanID}/messages/${msgID}/reactions/${reaction}`;
 module.exports.CHANNEL_MESSAGE_REACTION_USER = (chanID, msgID, reaction, userID) => `/channels/${chanID}/messages/${msgID}/reactions/${reaction}/${userID}`;
 module.exports.CHANNEL_MESSAGE_REACTIONS =                       (chanID, msgID) => `/channels/${chanID}/messages/${msgID}/reactions`;
+module.exports.CHANNEL_MESSAGE_REACTIONS_EMOJI =          (chanID, msgID, emoji) => `/channels/${chanID}/messages/${msgID}/reactions/${emoji}`;
```
Comment (abalabahaha) [inline lib/rest/Endpoints.js]: This duplicates line 12

**Review by abalabahaha — CHANGES_REQUESTED** [review]

**Comment by Khaaz** [comment]
Apparently you changed indentation of some files bringing this PR to 4k diff

**Review by abalabahaha — APPROVED** [review]

**Pull request merged**